src/library/components/Menu.js

```diff
@@ -0,0 +1,272 @@
+        menuWidth: width,
+        menuHeight: height,
+      },
+      () => {
+        Animated.parallel([
+          Animated.timing(this.state.menuSizeAnimation, {
+            toValue: { x: width, y: height },
+            duration: this.props.animationDuration,
+            easing: EASING,
+            useNativeDriver: false,
+          }),
+          Animated.timing(this.state.opacityAnimation, {
+            toValue: 1,
+            duration: this.props.animationDuration,
+            easing: EASING,
+            useNativeDriver: false,
+          }),
+        ]).start();
+      },
+    );
+  };
+
+  _onDismiss = () => {
+    if (this.props.onHidden) {
+      this.props.onHidden();
+    }
+  };
+
+  show = () => {
+    this._container.measureInWindow((left, top, buttonWidth, buttonHeight) => {
+      this.setState({
+        buttonHeight,
+        buttonWidth,
+        left,
+        menuState: STATES.SHOWN,
+        top,
+      });
+    });
+  };
+
+  hide = (onHidden) => {
+    Animated.timing(this.state.opacityAnimation, {
+      toValue: 0,
+      duration: this.props.animationDuration,
+      easing: EASING,
+      useNativeDriver: false,
+    }).start(() => {
+      // Reset state
+      this.setState(
+        {
+          menuState: STATES.HIDDEN,
+          menuSizeAnimation: new Animated.ValueXY({ x: 0, y: 0 }),
+          opacityAnimation: new Animated.Value(0),
+        },
+        () => {
+          if (onHidden) {
+            onHidden();
+          }
+
+          // Invoke onHidden callback if defined
+          if (Platform.OS !== 'ios' && this.props.onHidden) {
+            this.props.onHidden();
+          }
+        },
+      );
+    });
+  };
+
+  _hide = () => {
+    this.hide();
+  };
+
+  render() {
+    const { isRTL } = I18nManager;
+
+    const dimensions = Dimensions.get('window');
+    const { width: windowWidth } = dimensions;
+    const windowHeight = dimensions.height - (StatusBar.currentHeight || 0);
+
+    const {
+      menuSizeAnimation,
+      menuWidth,
+      menuHeight,
+      buttonWidth,
+      buttonHeight,
+      opacityAnimation,
+    } = this.state;
+    const menuSize = {
+      width: menuSizeAnimation.x,
+      height: menuSizeAnimation.y,
+    };
+
+    // Adjust position of menu
+    let { left, top } = this.state;
+    const transforms = [];
+
+    if (
+      (isRTL && left + buttonWidth - menuWidth > SCREEN_INDENT) ||
+      (!isRTL && left + menuWidth > windowWidth - SCREEN_INDENT)
+    ) {
+      transforms.push({
+        translateX: Animated.multiply(menuSizeAnimation.x, -1),
+      });
+
+      left = Math.min(windowWidth - SCREEN_INDENT, left + buttonWidth);
+    } else if (left < SCREEN_INDENT) {
+      left = SCREEN_INDENT;
+    }
+
+    // Flip by Y axis if menu hits bottom screen border
+    if (top > windowHeight - menuHeight - SCREEN_INDENT) {
+      transforms.push({
+        translateY: Animated.multiply(menuSizeAnimation.y, -1),
+      });
```

Comment:
perhaps, it'd help to introduce a helper function `flipX` and `flipY`, and then 
```suggestion
    const hitsBottomBorder = top > windowHeight - menuHeight - SCREEN_INDENT;
    if (hitsBottomBorder) {
      transforms.push({
        translateY: flipY(menuSize), <-- or menuSizeAnimation.y, but please see comment 🐃 
      });
```